sonarts-core/src/rules/noUnenclosedMultilineBlockRule.ts

```diff
@@ -0,0 +1,210 @@
+        if (i < array.length - 1) {
+          if (this.isConditionOrLoop(statement)) {
+            result.push({ prev: statement, next: array[i + 1] });
+          }
+        }
+        return result;
+      }, new Array<{ prev: ConditionOrLoop; next: ts.Statement }>())
+      .map(pair => {
+        return { topStatement: pair.prev, prev: this.extractLastBody(pair.prev), next: pair.next };
+      });
+  }
+
+  private isConditionOrLoop(statement: ts.Statement): statement is ConditionOrLoop {
+    return is(
+      statement,
+      ts.SyntaxKind.IfStatement,
+      ts.SyntaxKind.ForStatement,
+      ts.SyntaxKind.ForInStatement,
+      ts.SyntaxKind.ForOfStatement,
+      ts.SyntaxKind.WhileStatement,
+    );
+  }
+
+  private extractLastBody(statement: ConditionOrLoop): ts.Statement {
+    if (statement.kind === ts.SyntaxKind.IfStatement) {
+      if (!statement.elseStatement) {
+        return statement.thenStatement;
+      } else {
+        return statement.elseStatement;
+      }
+    } else {
+      return statement.statement;
+    }
+  }
+
+  private endAndStartPositions(pair: { prev: ts.Statement; next: ts.Statement }): Positions {
+    return {
+      prevStart: this.getLineAndCharacterOfPosition(pair.prev.getStart()),
+      prevEnd: this.getLineAndCharacterOfPosition(pair.prev.getEnd()),
+      nextStart: this.getLineAndCharacterOfPosition(pair.next.getStart()),
+      nextEnd: this.getLineAndCharacterOfPosition(pair.next.getEnd()),
+    };
+  }
+
+  private areAdjacent(positions: Positions): boolean {
+    return positions.prevEnd.line === positions.nextStart.line;
+  }
+
+  private raiseAdjacenceIssue(adjacentStatements: ChainedStatements) {
+    const conditional = adjacentStatements.topStatement.kind === ts.SyntaxKind.IfStatement;
+    this.addFailureAtNode(
+      adjacentStatements.next,
+      `This statement will not be executed ${conditional ? "conditionally" : "in a loop"}; ` +
+        `only the first statement will be. The rest will execute ${conditional ? "unconditionally" : "only once"}.`,
+    );
+  }
+
+  private haveSameIndentation(positions: Positions, chainedStatements: ChainedStatements): boolean {
+    return (
+      positions.prevStart.character === positions.nextStart.character &&
+      this.areMoreIndentedThanTopStatement(chainedStatements)
+    );
+  }
+
+  private areMoreIndentedThanTopStatement(chainedStatements: ChainedStatements): boolean {
+    const rootIndentation = this.getLineAndCharacterOfPosition(chainedStatements.topStatement.getStart()).character;
+    const prevIndentation = this.getLineAndCharacterOfPosition(chainedStatements.prev.getStart()).character;
+    return prevIndentation > rootIndentation;
+  }
+
+  private countStatementsInTheSamePile(reference: ts.Statement, statements: ts.Statement[]): number {
+    let startOfPile = this.getLineAndCharacterOfPosition(reference.getStart());
+    let lastLineOfPile = startOfPile.line;
+    for (let i = 0; i < statements.length; i++) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Expected a 'for-of' loop instead of a 'for' loop with this simple iteration [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4138)